subsys/nrf_security/src/drivers/cracen/cracenpsa/src/kmu.c

```diff
@@ -240,6 +242,74 @@ static bool can_sign(const psa_key_attributes_t *key_attr)
 	       (psa_get_key_usage_flags(key_attr) & PSA_KEY_USAGE_SIGN_HASH);
 }
 
+/**
+ * @brief Check provisioning state, and delete slots that were not completely provisioned.
+ *
+ * @return psa_status_t
+ */
+static psa_status_t verify_provisioning_state(void)
+{
+	uint32_t data;
+	int st = lib_kmu_read_metadata(PROVISIONING_SLOT, &data);
+
+	if (st == LIB_KMU_SUCCESS) {
+		uint32_t slot_id = data >> 8;
+		uint32_t num_slots = data & 0xff;
+
+		for (uint32_t i = 0; i < num_slots; i++) {
+			st = lib_kmu_revoke_slot(slot_id + i);
+			if (st != LIB_KMU_SUCCESS) {
+				if (!lib_kmu_is_slot_empty(slot_id + i)) {
+					return PSA_ERROR_HARDWARE_FAILURE;
+				}
+			}
+		}
+
+		st = lib_kmu_revoke_slot(PROVISIONING_SLOT);
+		if (st != LIB_KMU_SUCCESS) {
+			return PSA_ERROR_HARDWARE_FAILURE;
+		}
+	}
+	return PSA_SUCCESS;
```

Comment:
Here we will return success even if this functions returns LIB_KMU_ERROR, right? Maybe better to check for specific error codes?